st3/sublime_lib/show_selection_panel.py

```diff
@@ -0,0 +1,94 @@
+NOT_GIVEN = object()
+
+
+def show_selection_panel(
+    window,
+    items,
+    *,
+    flags=None,
+    labels=NOT_GIVEN,
+    selected=NOT_GIVEN,
+    on_select=None,
+    on_cancel=None,
+    on_highlight=None
+):
+    """Open a quick panel in the given window to select an item from a list.
+
+    :argument window: The :class:`sublime.Window` in which to show the panel.
+
+    :argument items: A :class:`list` of items to choose from. :exc:`ValueError`
+    will be raised if `items` is empty.
+
+    Optional keyword arguments:
+
+    :argument flags: A bitwise OR of :const:`sublime.MONOSPACE_FONT` and
+    :const:`sublime.KEEP_OPEN_ON_FOCUS_LOST`.
+
+    :argument labels: Either a list of labels or a function taking elements of
+    `items` to string labels. If `labels` is not given, the `items` themselves
+    will be taken as the `labels`.
+
+    Every label must be either a single value or a list of values.
+    :exc:`ValueError` will be raised if some labels are lists and others are not
+    or if the labels are lists with inconsistent lengths. Each value will be
+    converted to a string via :func:`str`.
+
+    :argument selected: The value in `items` that will be initially selected.
+    If `selected` is not given, no value will be initially selected.
+
+    :argument on_select: A callback accepting a value from `items` to be
+    invoked when the user chooses an item.
+
+    :argument on_cancel: A callback that will be invoked with no arguments if
+    the user closes the panel without choosing an item.
+
+    :argument on_highlight: A callback accepting a value from `items` to be
+    invoked every time the user changes the highlighted item in the panel.
+    """
```

Comment:
Add `:raises:` to docstrings? Doesn't need to be as detailed as to include every individual exception, but should at least mention all exception types. Also consider the exception that `list.index` throws.